stream_alert/alert_processor/outputs.py

```diff
@@ -803,3 +805,272 @@ def dispatch(self, **kwargs):
+            ('username',
+             OutputProperty(description='the Jira username',
+                            mask_input=True,
+                            cred_requirement=True)),
+            ('password',
+             OutputProperty(description='the Jira password',
+                            mask_input=True,
+                            cred_requirement=True)),
+            ('url',
+             OutputProperty(description='the Jira url',
+                            mask_input=True,
+                            cred_requirement=True)),
+            ('project_key',
+             OutputProperty(description='the Jira project key',
+                            mask_input=False,
+                            cred_requirement=True)),
+            ('issue_type',
+             OutputProperty(description='the Jira issue type',
+                            mask_input=False,
+                            cred_requirement=True)),
+            ('aggregate',
+             OutputProperty(description='the Jira aggregation behavior to aggregate '
+                                        'alerts by rule name (yes/no)',
+                            mask_input=False,
+                            cred_requirement=True))
+        ])
+
+    def _search_jira(self, jql, max_results=100, validate_query=True, fields=None):
+        """Search Jira for issues using a JQL query
+
+        Args:
+            jql (str): The JQL query
+            max_results (int): Maximum number of results to return
+            validate_query (bool): Whether to validate the JQL query or not
+            fields (list): List of fields to return for each issue
+        Returns:
+            list: list of issues matching JQL query
+        """
+        search_url = os.path.join(self.base_url, self.SEARCH)
+        headers = {'Content-Type': 'application/json', 'cookie' : self.auth_cookie}
+        params = {'jql': jql,
+                  'maxResults': max_results,
+                  'validateQuery': validate_query,
+                  'fields': fields}
+        issues = []
+
+        resp = self._get_request(search_url, params=params, headers=headers, verify=False)
+
+        try:
+            issues = resp.json()['issues']
+        except (ValueError, KeyError):
+            LOGGER.error('An error occurred while searching Jira [%s] : [%s]',
+                         resp.status_code,
+                         resp.content)
+
+        return issues
+
+    def _create_comment(self, issue_id, comment):
+        """Add a comment to an existing issue
+
+        Args:
+            issue_id (str): The existing issue ID or key
+            comment (str): The body of the comment
+
+        Returns:
+            int: ID of the created comment or False if unsuccessful
+        """
+
+        comment_url = os.path.join(self.base_url, self.COMMENT.format(issue_id))
+        comment_body = {'body' : comment}
+        headers = {'Content-Type': 'application/json', 'cookie' : self.auth_cookie}
```

Comment:
Suggestion: there's a good amount of duplication here with `headers = ..` code. I'd suggest consolidating a bit with something along the lines of:

```
@output
class JiraOutput(StreamOutputBase):
    DEFAULT_HEADERS = {'Content-Type': 'application/json'}

    @classmethod
    def _get_default_headers(cls):
        """Class method to be used when you do not need the 'cookie' param"""
        return cls.DEFAULT_HEADERS.copy()

    def _get_headers(self):
        """Instance method to be used when you want to combine the cookie with the default headers"""
        return dict(self._get_default_headers(), **{'cookie': self.auth_cookie})
```
This also has the benefit of the headers being a class constant with less chance of a typo bringing things down.